crates/sui-core/src/authority.rs

```diff
@@ -866,6 +870,84 @@ impl AuthorityState {
             .check_owned_object_locks_exist(owned_object_refs)
     }
 
+    /// Make sure not to leak any private info here
+    pub(crate) async fn state_dump(
+        &self,
+        tx_digest: &TransactionDigest,
+        effects: &TransactionEffects,
+        expected_effects_digest: TransactionEffectsDigest,
+        inner_temporary_store: &InnerTemporaryStore,
+        certificate: &VerifiedExecutableTransaction,
+        path: &Path,
+    ) -> SuiResult<PathBuf> {
+        // Epoch info
+        let epoch_store = self.load_epoch_store_one_call_per_task();
+        let executed_epoch = epoch_store.epoch();
+        let reference_gas_price = epoch_store.reference_gas_price();
+        let epoch_start_config = epoch_store.epoch_start_config();
+        let protocol_version = epoch_store.protocol_version().as_u64();
+        let epoch_start_timestamp_ms = epoch_start_config.epoch_data().epoch_start_timestamp();
+
+        // Record all system packages at this version
+        let mut relevant_system_packages = Vec::new();
+        for sys_package_id in BuiltInFramework::all_package_ids() {
+            if let Some(w) = self.get_object(&sys_package_id).await? {
+                relevant_system_packages.push(w)
+            }
+        }
+
+        // Record all the shared objects
+        let mut shared_objects = Vec::new();
+        for (id, ver, _) in effects.shared_objects() {
+            if let Some(w) = self.database.get_object_by_key(id, *ver)? {
+                shared_objects.push(w)
```

Comment:
Are these not going to be covered in runtime_read_objects?
Similar question for loaded_child_objects

Reply:
`runtime_read_objects`
Not always/ In some cases dependent packages are fetched even thought they're not in the input objects
`loaded_child_objects `
When a child object is just read but not mutated, the object does not show up anywhere else. Had to add this logic explicitly for debugging weeks back.